lighthouse-core/test/gather/gatherers/image-elements-test.js

```diff
@@ -0,0 +1,282 @@
+      },
+    ];
+
+    const index = gatherer.indexNetworkRecords(networkRecords);
+
+    expect(index).toEqual({
+      'https://example.com/img.avif': {
+        finished: true,
+        mimeType: 'application/octect-stream',
+        statusCode: 200,
+        url: 'https://example.com/img.avif',
+      },
+      'https://example.com/img.png': {
+        finished: true,
+        mimeType: 'image/png',
+        statusCode: 200,
+        url: 'https://example.com/img.png',
+      },
+      'https://example.com/img.webp': {
+        finished: true,
+        mimeType: 'application/octect-stream',
+        statusCode: 200,
+        url: 'https://example.com/img.webp',
+      },
+    });
+  });
+
+  it('ignores bad status codes', () => {
+    const gatherer = mockImageElements();
+    const networkRecords = [
+      {
+        mimeType: 'image/png',
+        url: 'https://example.com/img.png',
+        finished: true,
+        statusCode: 200,
+      },
+      {
+        mimeType: 'application/octect-stream',
+        url: 'https://example.com/img.webp',
+        finished: false,
+      },
+      {
+        mimeType: 'application/octect-stream',
+        url: 'https://example.com/img.avif',
+        finished: true,
+        statusCode: 404,
+      },
+    ];
+
+    const index = gatherer.indexNetworkRecords(networkRecords);
+
+    expect(index).toEqual({
+      'https://example.com/img.png': {
+        finished: true,
+        mimeType: 'image/png',
+        statusCode: 200,
+        url: 'https://example.com/img.png',
+      },
+    });
+  });
+});
+
+describe('.collectExtraDetails', () => {
+  let gatherer = mockImageElements();
+
+  beforeEach(() => {
+    gatherer = mockImageElements();
+    gatherer.fetchSourceRules.mockImplementation();
+    gatherer.fetchElementWithSizeInformation.mockImplementation();
+  });
+
+  it('times out', () => {
+    const elements = [
+      {node: {}, isInShadowDOM: false, isCss: false},
+      {node: {}, isInShadowDOM: false, isCss: false},
+      {node: {}, isInShadowDOM: false, isCss: false},
+    ];
+    gatherer.fetchSourceRules.mockImplementation(async () => {
+      jest.advanceTimersByTime(6000);
+    });
+
+    gatherer.collectExtraDetails({}, elements, {});
+
+    expect(gatherer.fetchSourceRules).toHaveBeenCalledTimes(1);
+  });
+
+  it('fetch source rules for unsized images', () => {
+    const elements = [
+      {node: {}, isInShadowDOM: false, isCss: false},
+      {node: {}, isInShadowDOM: true, isCss: false},
+      {node: {}, isInShadowDOM: false, isCss: true},
+    ];
+
+    gatherer.collectExtraDetails({}, elements, {});
+
+    expect(gatherer.fetchSourceRules).toHaveBeenCalledTimes(1);
+    expect(gatherer.fetchSourceRules).toHaveBeenCalledWith(undefined, undefined, elements[0]);
+  });
+
+  it('fetch size information for image with picture', () => {
+    const elements = [
+      {node: {}, src: 'https://example.com/a.png', isPicture: true, isCss: true, srcset: 'src'},
```

Comment:
FWIW `isPicture` and `isCss` should never be true at the same time.

Reply:
I'm pretty sure this function is bugged. `networkRecord` will always be true because `{}` is truthy.